### PR TITLE
Theming the new filter bar

### PR DESCRIPTION
**Pull request by AMBacelar**
## Who is this for?
Users of the new search pages

## What is it doing for them?
Gives them a shiny new filter bar to use and mess around with!

### CODE REVIEW

File: common/views/components/SearchFilters/SearchFilters.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ type Props = {
   changeHandler: () => void;
   filters: Filter[];
   linkResolver: (params: ParsedUrlQuery) => LinkProps;
+  newStyle?: boolean;
```
Comment (rcantin-w) [inline common/views/components/SearchFilters/SearchFilters.tsx:16]: Could we rename to `isNewStyle` to enforce that it's boolean?

File: common/views/components/SearchTabs/SearchTabs.tsx [hunk 2]
```diff
@@ -258,7 +258,7 @@ const SearchTabs: FunctionComponent<Props> = ({
             }}
             ariaDescribedBy="images-form-description"
             isImageSearch={true}
-            shouldShowFilters={isEnhanced && shouldShowFilters} // non js images filters doesnt work hide for now\
+            shouldShowFilters={shouldShowFilters} // non js images filters doesnt work hide for now\
```
Comment (rcantin-w) [inline common/views/components/SearchTabs/SearchTabs.tsx:261]: What does "for now" mean in this case (should we create a ticket/this be a TODO comment?)

File: common/views/components/ModalMoreFilters/ModalMoreFilters.tsx [hunk 3]
```diff
@@ -129,6 +137,78 @@ const CheckboxFilter = ({ f, changeHandler, form }: CheckboxFilterProps) => {
...
+            changeHandler();
+          }
+        }}
+        form={form}
+      />
+    </fieldset>
+  );
+};
+
+const PaletteColorPicker = dynamic(
```
Comment (rcantin-w) [inline common/views/components/ModalMoreFilters/ModalMoreFilters.tsx:191]: For my own knowledge; why does this needs to be a dynamic import? (and should we leave a comment to explain it to future devs too)

File: common/views/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ type Props = {
   openButtonRef?: MutableRefObject<HTMLElement | null>;
   removeCloseButton?: boolean;
   showOverlay?: boolean;
-  modalStyle?: 'filters' | 'calendar' | 'default';
+  modalStyle?: 'filters' | 'calendar' | 'default' | 'filters-new';
```
Comment (rcantin-w) [inline common/views/components/Modal/Modal.tsx:37]: super nitpick; I'd leave default to be the last one.
Although having said that I looked at the switch/case using it 

```
function determineModal(modalStyle: Props['modalStyle']) {
  switch (modalStyle) {
    case 'filters':
      return FiltersModal;
    case 'filters-new':
      return FiltersModalNew;
    case 'calendar':
      return CalendarModal;
    default:
      return BaseModalWindow;
  }
}
```

and it seems like it just ignores the 'default' string (going to default anyway). Is 'default' specified anywhere? Could it just be removed from the allowed types?

File: catalogue/webapp/pages/works.tsx [hunk 5]
```diff
@@ -106,7 +106,14 @@ const Works: NextPage<Props> = ({ works, worksRouteProps }) => {
...
         jsonLd={{ '@type': 'WebPage' }}
         siteSection="collections"
-        // TODO: Why do we remove the 'main' role here?
+        /**
+         * We explicitly set the main role to be the container of the search results
+         * this is for accessibility purposes e.g. screen readers/keyboard only users,
+         * who would use the 'skip to main' element, and
+         * would expect to be on the first search result.
+         *
+         * this behaviour matches the expectations set by Google
```
Comment (rcantin-w) [inline catalogue/webapp/pages/works.tsx:115]: Thanks for that!

File: catalogue/webapp/pages/search/images.tsx [hunk 6]
```diff
@@ -216,6 +217,9 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
         }),
       };
     }
+    if (params.color) {
+      params.color = params.color.replace('#', '');
```
Comment (rcantin-w) [inline catalogue/webapp/pages/search/images.tsx:221]: Why do we do this again? I think you explained to me at the office...

File: catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx [hunk 7]
```diff
@@ -115,7 +115,30 @@ const SearchLayout: FunctionComponent<{ hasEventsExhibitions: boolean }> = ({
...
+      }
+      formValues['production.dates.from'] = from;
+      for (let i = 0; i < formValues['production.dates.to'].length; i++) {
+        const element = formValues['production.dates.to'][i];
+        if (element.trim() !== '') {
+          to = element.trim();
+        }
+      }
+      formValues['production.dates.to'] = to;
+    }
```
Comment (rcantin-w) [inline catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx:141]: Could we leave a comment to explain the above? At Alex's suggestion on my PR review, I created a `catalogue/webapp/utils/search.ts` file too for Sort functions, where we also tried to leave comments explain various logic, might be worth adding that function there? There's a Filters section at the bottom!

**Comment by rcantin-w** [comment]
Few visual suggestions:

Could we align it to the left? And make it look like the sort dropdown? (black border, smaller text)
Screenshot 2022-12-06 at 14 41 59


Running it locally, I'm not seeing the All Filters buttons right away, but when I resize the window it appears. I also ended up with an error, here's a screen recording: [https://we.tl/t-pQl0metNMg](https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwe.tl%2Ft-pQl0metNMg%3Futm_campaign%3DTRN_TDL_05%26utm_source%3Dsendgrid%26utm_medium%3Demail%26trk%3DTRN_TDL_05&data=05%7C01%7Cr.cantin%40wellcome.org%7C475f640b6d574df2609008dad798fd28%7C3b7a675a1fc84983a100cc52b7647737%7C0%7C0%7C638059349327695629%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C2000%7C%7C%7C&sdata=auF5bEoE%2FH64WoiS%2BoOw5NdxCHXtWyUpwTT2J1L1nJ8%3D&reserved=0)

Happy to jump on a call to discuss, seems like a real puzzle of a component, and it already looks so much better! 🙌

File: common/views/components/SearchFilters/SearchFilters.tsx (hunk 1 above)
Comment (AMBacelar) [reply common/views/components/SearchFilters/SearchFilters.tsx]: done

File: common/views/components/ModalMoreFilters/ModalMoreFilters.tsx (hunk 3 above)
Comment (AMBacelar) [reply common/views/components/ModalMoreFilters/ModalMoreFilters.tsx]: I've made the change

File: common/views/components/Modal/Modal.tsx (hunk 4 above)
Comment (AMBacelar) [reply common/views/components/Modal/Modal.tsx:37]: done

File: catalogue/webapp/pages/search/images.tsx (hunk 6 above)
Comment (AMBacelar) [reply catalogue/webapp/pages/search/images.tsx:221]: I'll add a comment to explain moving forward

File: common/views/components/Modal/Modal.tsx [hunk 8]
```diff
@@ -165,6 +165,63 @@ const FiltersModal = styled(BaseModalWindow).attrs<BaseModalProps>({
   padding-right: 0px;
 `;
 
+const FiltersModalNew = styled(Space).attrs<BaseModalProps>({
```
[the remark this answers is not in the document]
Comment (AMBacelar) [reply common/views/components/Modal/Modal.tsx:168]: done

File: common/views/components/ButtonSolid/ButtonSolid.tsx [hunk 9]
```diff
@@ -185,6 +187,7 @@ export const SolidButton = styled(BaseButton).attrs<SolidButtonProps>(
       }`;
     } else {
       return `
+        border-radius: ${props.isPill ? '24px' : '0'};
```
[the remark this answers is not in the document]
Comment (AMBacelar) [reply common/views/components/ButtonSolid/ButtonSolid.tsx]: done

File: common/views/components/SearchFilters/SearchFiltersDesktop.tsx [hunk 10]
```diff
@@ -13,62 +14,77 @@ import Icon from '../Icon/Icon';
...
   ColorFilter as ColorFilterType,
   filterLabel,
+  Filter,
 } from '../../../services/catalogue/filters';
 import ModalMoreFilters from '../ModalMoreFilters/ModalMoreFilters';
 import { ResetActiveFilters } from './ResetActiveFilters';
 import ButtonSolid, { ButtonTypes } from '../ButtonSolid/ButtonSolid';
 import { filter } from '@weco/common/icons';
 import { themeValues } from '@weco/common/views/themes/config';
+import { useLayoutEffect } from 'react';
```
Comment (rcantin-w) [inline common/views/components/SearchFilters/SearchFiltersDesktop.tsx:30]: We already have a `from react` import at the top of the document. I use an ESLint extension in VSCode that picks up whatever our config is and automatically makes changes on save, it's great for this sort of thing: //marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

File: catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx (hunk 7 above)
Comment (AMBacelar) [reply catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx]: no longer needed!
So this code is gone

**Comment by alexwlchan** [comment]
Looking at this now.

File: common/views/components/SearchTabs/SearchTabs.tsx (hunk 2 above)
Comment (alexwlchan) [reply common/views/components/SearchTabs/SearchTabs.tsx:261]: I thought this comment was relating to the `isEnhanced` variable, and if Adi has got images filters work without JavaScript, can we delete it?

File: common/views/components/PaletteColorPicker/PaletteColorPicker.tsx [hunk 11]
```diff
@@ -162,32 +165,56 @@ const PaletteColorPicker: FunctionComponent<Props> = ({
...
-        </ClearButton>
-      </TextWrapper>
+      <noscript>
+        <input
+          form={form}
+          type="color"
+          name={name}
+          value={
+            color
+              ? color[0] === '#' || color[0] === '%'
```
Comment (alexwlchan) [inline common/views/components/PaletteColorPicker/PaletteColorPicker.tsx:175]: Why might colour by `%`?

**Review by alexwlchan — APPROVED** [review]
Overall looks really nice, a big improvement on what went before. A couple of small comments inline, but the code looks solid and you've solved the big problems. 💪 

I'm getting an error when I run it locally, is this significant?

```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at SearchFiltersDesktop (webpack-internal:///../../common/views/components/SearchFilters/SearchFiltersDesktop.tsx:250:3)
    at SearchFilters (webpack-internal:///../../common/views/components/SearchFilters/SearchFilters.tsx:29:3)
    at div
    at CatalogueSearchPage (webpack-internal:///./pages/search/works.tsx:86:3)
    at WithSuperJSON (/Users/alexwlchan/repos/dotorg/node_modules/babel-plugin-superjson-next/src/tools.tsx:61:22)
    at div
    at div
    at PageLayoutComponent (webpack-internal:///../../common/views/components/PageLayout/PageLayout.tsx:65:3)
    at GlobalInfoBarContextProvider (webpack-internal:///../../common/views/components/GlobalInfoBarContext/GlobalInfoBarContext.tsx:20:3)
    at PageLayout
    at CataloguePageLayout (webpack-internal:///./components/CataloguePageLayout/CataloguePageLayout.tsx:39:5)
    at SearchLayout (webpack-internal:///./components/SearchPageLayout/SearchPageLayout.tsx:61:3)
    at OutboundLinkTracker (webpack-internal:///../../common/views/components/OutboundLinkTracker/OutboundLinkTracker.tsx:19:3)
    at exports.ThemeProvider (/Users/alexwlchan/repos/dotorg/node_modules/styled-components/src/models/ThemeProvider.js:48:22)
    at AppContextProvider (webpack-internal:///../../common/views/components/AppContext/AppContext.tsx:44:3)
    at UserProvider (webpack-internal:///../../common/views/components/UserProvider/UserProvider.tsx:30:3)
    at ApmContextProvider (webpack-internal:///../../common/views/components/ApmContext/ApmContext.tsx:23:3)
    at WecoApp (webpack-internal:///../../common/views/pages/_app.tsx:77:3)
    at SearchContextProvider (webpack-internal:///../../common/views/components/SearchContext/SearchContext.tsx:26:3)
    at CatalogueApp
    at ae (/Users/alexwlchan/repos/dotorg/node_modules/styled-components/src/models/StyleSheetManager.js:34:48)
    at AppContainer (/Users/alexwlchan/repos/dotorg/node_modules/next/server/render.tsx:521:27)
```

I have a couple of other notes, but not blockers:

* When there aren't any filters available, we get a sad and empty popup. It might be worth adding some text "(No filters available)" to explain what's going on, but that's also broken on the existing version.
* The layout moves around a bit when I open a filter for the first time, looks like waiting for fonts to snap in?
* If I click "Reset filters" in the list of active filters, I'm returned to new search; if I click it in the sidebar, I go to old search.

None of these are a blocker and I think it'd be good to get this merged and in front of Jonathan for his feedback before the end of the year, so just a couple of inline comments and then we can make this live behind a toggle. 🥳

File: common/views/components/PaletteColorPicker/PaletteColorPicker.tsx (hunk 11 above)
Comment (AMBacelar) [reply common/views/components/PaletteColorPicker/PaletteColorPicker.tsx:175]: url encoded value for the hash symbol will begin with the % symbol

**Pull request merged**